dist/init/linux-systemd/caddy.service

```diff
@@ -33,16 +38,25 @@ PrivateDevices=true
 ProtectHome=true
 ; Make /usr, /boot, /etc and possibly some more folders read-only.
 ProtectSystem=full
-; … except /etc/ssl/caddy, because we want Letsencrypt-certificates there.
+; … except /etc/ssl/caddy, because we want Letsencrypt-certificates there
+;   and /var/log/caddy, because we want a place where logs can go.
 ;   This merely retains r/w access rights, it does not add any new. Must still be writable on the host!
-ReadWriteDirectories=/etc/ssl/caddy
+ReadWritePaths=/etc/ssl/caddy /var/log/caddy
```

Comment:
You renamed `ReadWriteDirectories` to `ReadWritePaths`. The latter is available from v231 and later.

One of the most popular Linux distributions is Ubuntu, and its latest LTS release ships with v229. Although one can update it easily (it interfaces with a wide span of dep libs), most users don't know how and/or don't bother to. I suggest we go with what comes with v229 for now.  
http://packages.ubuntu.com/xenial/systemd